api/v1/openapi.yaml

```diff
@@ -2418,6 +2502,98 @@ definitions:
     properties:
       realized:
         "$ref": "#/definitions/ServiceSpec"
+  Recorder:
+    description: Collection of wildcard filters for pcap recorder
+    type: object
+    properties:
+      spec:
+        "$ref": "#/definitions/RecorderSpec"
+      status:
+        "$ref": "#/definitions/RecorderStatus"
+  RecorderMask:
+    description: Individual mask for pcap recorder
+    type: object
+    properties:
+      status:
+        "$ref": "#/definitions/RecorderMaskStatus"
+  RecorderFilter:
+    description: n-tuple filter to match traffic to be recorded
+    type: object
+    properties:
+      src-prefix:
+        description: Layer 3 source CIDR
+        type: string
+      src-port:
+        description: Layer 4 source port, zero (or in future range)
+        type: string
+      dst-prefix:
+        description: Layer 3 destination CIDR
+        type: string
+      dst-port:
+        description: Layer 4 destination port, zero (or in future range)
+        type: string
+      protocol:
+        description: Layer 4 protocol
+        type: string
+        enum:
+        - TCP
+        - UDP
+        - ANY
```

Comment:
Can we also support ICMP/ICMPv6? The proposed Hubble API and the `u8proto` proto package already assume they work, but I'm less certain about the datapath.

Reply:
For the load balancer itself, it won't be able to IPIP encap ICMP traffic. The LB currently only supports UDP+TCP. I think once we generalize the pcap recorder further, we could integrate it.